src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -985,71 +1084,45 @@ private int constructMergedTable(Table table, List<FeedToMerge> feedsToMerge,
-                "Note: \"special\" stops with location_type > 0 need not specify this value " +
-                "(%d special stops found in feed).",
-            stopsMissingStopCodeCount,
-            stopsCount,
-            specialStopsCount
-        );
-    }
-
-    /** Get the set of shared fields for all feeds being merged for a specific table. */
-    private Set<Field> getSharedFields(List<FeedToMerge> feedsToMerge, Table table) throws IOException {
-        Set<Field> sharedFields = new HashSet<>();
-        // First, iterate over each feed to collect the shared fields that need to be output in the merged table.
-        for (FeedToMerge feed : feedsToMerge) {
-            CsvReader csvReader = table.getCsvReader(feed.zipFile, null);
-            // If csv reader is null, the table was not found in the zip file.
-            if (csvReader == null) {
-                continue;
+    private static MergeStrategy getMergeStrategy(List<FeedToMerge> feedsToMerge) throws IOException {
+        // Iterate over both feeds (future feed is first).
+        for (int i = 0; i < feedsToMerge.size(); i++) {
+            FeedToMerge feed = feedsToMerge.get(i);
+            Set<Table> tablesToCheck = Sets.newHashSet(Table.TRIPS, Table.CALENDAR, Table.CALENDAR_DATES);
+            for (Table table : tablesToCheck) {
+                feed.idsForTable.get(table).addAll(getIdsForTable(feed.zipFile, table));
             }
-            // Get fields found from headers and add them to the shared fields set.
-            Field[] fieldsFoundInZip = table.getFieldsFromFieldHeaders(csvReader.getHeaders(), null);
-            sharedFields.addAll(Arrays.asList(fieldsFoundInZip));
         }
-        return sharedFields;
-    }
-
-    /**
-     * Checks whether a collection of fields contains a field with the provided name.
-     */
-    private boolean containsField(Collection<Field> fields, String fieldName) {
-        for (Field field : fields) if (field.name.equals(fieldName)) return true;
-        return false;
-    }
-
-    /** Checks that any of a set of errors is of the type {@link NewGTFSErrorType#DUPLICATE_ID}. */
-    private boolean hasDuplicateError(Set<NewGTFSError> errors) {
-        for (NewGTFSError error : errors) {
-            if (error.errorType.equals(NewGTFSErrorType.DUPLICATE_ID)) return true;
+        // We are on the last (second) feed to merge. Check for conflicts with future feed.
+        FeedToMerge futureFeed = feedsToMerge.get(0);
+        FeedToMerge currentFeed = feedsToMerge.get(1);
+        Set<String> pastTripIds = currentFeed.idsForTable.get(Table.TRIPS);
+        Set<String> futureTripIds = futureFeed.idsForTable.get(Table.TRIPS);
+        Set<String> pastServiceIds = new HashSet<>();
+        pastServiceIds.addAll(currentFeed.idsForTable.get(Table.CALENDAR));
+        pastServiceIds.addAll(currentFeed.idsForTable.get(Table.CALENDAR_DATES));
```

Comment:
Is it past or current. It'd be really helpful to pick one thing to call this thing to avoid confusion.